server/service/schedule/schedule_test.go

```diff
@@ -298,3 +301,247 @@ func TestJobPanicRecover(t *testing.T) {
 		t.Error("timeout")
 	}
 }
+
+func TestScheduleReleaseLock(t *testing.T) {
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	defer cancelFunc()
+
+	start := time.Now()
+
+	ds := new(mock.Store)
+	var mockLock struct {
+		owner  string
+		expiry time.Time
+		count  int
+
+		mu sync.Mutex
+	}
+
+	ds.LockFunc = func(ctx context.Context, name string, owner string, expiration time.Duration) (bool, error) {
+		mockLock.mu.Lock()
+		defer mockLock.mu.Unlock()
+
+		now := time.Now()
+		if mockLock.owner == owner || now.After(mockLock.expiry) {
+			mockLock.owner = owner
+			mockLock.expiry = now.Add(expiration)
+			mockLock.count = mockLock.count + 1
+
+			return true, nil
+		}
+		return false, nil
+	}
+
+	unlock := make(chan int)
+	ds.UnlockFunc = func(context.Context, string, string) error {
+		unlock <- 1
```

Comment:
Not something that needs changing, but another way to do this would be to send `struct{}{}`.

Reply:
Yeah, I actually refactored that in my WIP follow up to this PR :)